prog/heartbeat.rb

```diff
@@ -3,8 +3,14 @@
 require "excon"
 
 class Prog::Heartbeat < Prog::Base
+  CONNECTED_APPLICATION_QUERY = <<SQL
+SELECT count(DISTINCT application_name)
+FROM pg_stat_activity
+WHERE application_name ~ '^(bin/respirate|bin/monitor|.*/puma)$'
+SQL
+
   label def wait
-    DB["SELECT 1"].first
+    nap 10 unless DB.get(CONNECTED_APPLICATION_QUERY) == 3
```

Comment:
Should we add a log here to identify what's missing? For instance, when the `monitor` is down, we receive a "Respirate site is down" page. It could be useful to have a log indicating that the respirate is operational, but the `monitor` is down.

Reply:
Alright, added a second patch to do it that way. I think I will leave them separate for comparison sake, even after application.